python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -337,6 +351,37 @@ def __reduce__(self):
             self._reader_ref,
         )
 
+    def resize_channel_if_needed(self, serialized_value):
+        # serialized_value.total_bytes *only* includes the size of the data. It does not
+        # include the size of the metadata, so we must account for the size of the
+        # metadata explicitly.
+        size = serialized_value.total_bytes + len(serialized_value.metadata)
+        if size > self._typ.buffer_size_bytes:
+            # Now make the channel backing store larger.
+            self._typ.buffer_size_bytes = size
+            # TODO(jhumphri): I assume the current writer ref object is automatically
+            # freed when the reference to it is destroyed in the line below?
```

Comment:
I don't think it will get automatically freed. Can you change this to a TODO to handle garbage collection?

Reply:
Done.